### PR TITLE
Handle already-existing symlinks without error

### PR DESCRIPTION
**Pull request by tylerwilliams**
If a symlink exists with the correct target; use it, and if the target is incorrect, attempt to remove it and recreate.

### CODE REVIEW

File: enterprise/server/remote_execution/dirtools/dirtools.go [hunk 1]
```diff
@@ -928,6 +945,24 @@ func DownloadTree(ctx context.Context, env environment.Env, instanceName string,
...
+					// If the symlink exists and is correct;
+					// skip the error.
+					if checkSymlink(symlinkNode.GetTarget(), nodeAbsPath) {
+						continue
+					}
+					// If the symlink exists but was
+					// incorrect; try to remove it
+					if err := removeSymlink(nodeAbsPath); err != nil {
+						return err
+					}
```
Comment (bduffany) [inline enterprise/server/remote_execution/dirtools/dirtools.go:958]: wdyt about replacing these two things w/ `os.RemoveAll(nodeAbsPath)`? This way even if the existing path is not a symlink we can just obliterate it and place the correct symlink on top of it.

Comment (bduffany) [reply enterprise/server/remote_execution/dirtools/dirtools.go:958]: even if a directory already exists at the path, we ultimately want to replace it with a symlink, so I think RemoveAll() is probably the right thing to do.

Also I wasn't sure so had to double check - if the existing path is a symlink to another directory, RemoveAll() does not remove the directory that it points to. It just removes the link.

Comment (tylerwilliams) [reply enterprise/server/remote_execution/dirtools/dirtools.go]: yeah same haha just double checked that. anyway this is Done.

**Review by siggisim — APPROVED** [review]

File: enterprise/server/remote_execution/dirtools/dirtools_test.go [hunk 2]
```diff
@@ -722,6 +722,180 @@ func TestDownloadTreeEmptyDigest(t *testing.T) {
...
+				Digest: fileADigest,
+			},
+		},
+		Symlinks: []*repb.SymlinkNode{
+			&repb.SymlinkNode{
+				Name:   "fileA.symlink",
+				Target: "./fileA.txt",
+			},
+		},
+	}
```
Comment (bduffany) [inline enterprise/server/remote_execution/dirtools/dirtools_test.go:818]: nit: the `childDir` makes the test slightly hard to follow - if you wanted you could have this be the top-level dir.

File: enterprise/server/remote_execution/dirtools/dirtools_test.go [hunk 3]
```diff
@@ -722,6 +722,180 @@ func TestDownloadTreeEmptyDigest(t *testing.T) {
...
+	info, err := dirtools.DownloadTree(ctx, env, "", repb.DigestFunction_SHA256, directory, tmpDir, &dirtools.DownloadTreeOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NotNil(t, info, "transfers are not nil")
+	assert.Equal(t, int64(0), info.FileCount, "zero files were transferred")
+	assert.DirExists(t, filepath.Join(tmpDir, "my-directory"), "my-directory should exist")
+	assert.FileExists(t, filepath.Join(tmpDir, "my-directory/fileA.txt"), "fileA.txt should exist")
+	assert.FileExists(t, filepath.Join(tmpDir, "fileB.txt"), "fileB.txt should exist")
```
Comment (bduffany) [inline enterprise/server/remote_execution/dirtools/dirtools_test.go:896]: nit: add an assertion about the new symlink target for fileA.symlink?

**Review by bduffany — APPROVED** [review]

File: enterprise/server/remote_execution/dirtools/dirtools_test.go (hunk 3 above)
Comment (tylerwilliams) [reply enterprise/server/remote_execution/dirtools/dirtools_test.go]: Done.

File: enterprise/server/remote_execution/dirtools/dirtools_test.go (hunk 2 above)
Comment (tylerwilliams) [reply enterprise/server/remote_execution/dirtools/dirtools_test.go]: Done.
